components/places/src/ffi.rs

```diff
@@ -379,6 +401,27 @@ impl FrecencyThresholdOption {
     }
 }
 
+// We define those types to cross the FFI
```

Comment:
This is the dance I mentioned in the PR comment

One alternative is to:
- Change the variant for the internal `MatchReason` from `Url` to another name
- Change the `Tags` internal variant to not have associated data (or change `MatchUrl` to use our `.udl` syntax for enums with associated data, although... AC doesn't use the data so...)
- Then expose the internal SearchResult and MatchReason as apposed defining new types here

Thoughts about the tradeoff here?

Reply:
This seems OK, although I think it's a uniffi bug that we should open and even try to fix. The comment in the code isn't quite clear about why renaming `Url` is a solution (ie, the comment doesn't describe the problem)

The fact `tags` has associated data but other variants do not seems quite odd and as you seem to be implying it's not actually used by Kotlin - and I can't really see how it would be helpful - so dropping that tag seems an easy solution, but I'm fine either way.

Reply:
filed https://github.com/mozilla/uniffi-rs/issues/1137 for the uniffi bug, I'll also file one of those follow-ups for the tags to see if there's a better approach there (like maybe just drop the associated data from `Tags` in the internal type as well, or remove that variant as a whole)